src/com/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -1094,48 +1094,70 @@
     (dissoc row-map :environment_id)
     row-map))
 
+(defn normalize-resource-event
+  "Prep `event` for comparison/computation of a hash"
+  [event]
+  (-> event
+      (utils/update-when [:timestamp] to-timestamp)
+      (utils/update-when [:old-value] sutils/db-serialize)
+      (utils/update-when [:new-value] sutils/db-serialize)
+      (assoc :containing-class (find-containing-class (:containment-path event)))))
+
+(defn normalize-report
+  "Prep the report for comparison/computation of a hash"
+  [report]
+  (-> report
+      (update-in [:start-time] to-timestamp)
+      (update-in [:end-time] to-timestamp)
+      (update-in [:resource-events] #(map normalize-resource-event %))))
```

Comment:
Could be `(partial map normalize-resource-event)`, if we like point-free style. Me, I like what you have here.